frameworks/kafka/universe/config.json

```diff
@@ -187,23 +187,118 @@
               }
             },
             "transport_encryption": {
+              "title": "Transport Encryption Configuration",
               "type": "object",
-              "description": "Transport encryption settings",
+              "description": "Transport Encryption Settings",
               "properties": {
                 "enabled": {
                   "description": "Enable transport encryption (TLS)",
                   "type": "boolean",
+                  "enum": [
+                    true,
+                    false
+                  ],
                   "default": false
-                },
-                "allow_plaintext": {
-                  "description": "Allow plaintext alongside encrypted traffic",
-                  "type": "boolean",
-                  "default": false
-                },
-                "ciphers": {
-                  "description": "Comma-separated list of JSSE Cipher Suite Names",
-                  "type": "string",
-                  "default": "TLS_RSA_WITH_AES_128_GCM_SHA256,TLS_RSA_WITH_AES_128_CBC_SHA256,TLS_RSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_256_CBC_SHA256,TLS_DHE_RSA_WITH_AES_128_GCM_SHA256,TLS_DHE_RSA_WITH_AES_128_CBC_SHA256,TLS_DHE_RSA_WITH_AES_256_GCM_SHA384,TLS_DHE_RSA_WITH_AES_256_CBC_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384"
+                }
+              },
+              "dependencies": {
```

Comment:
@rishabh96b can you explain why we need this?

Reply:
I've added custom tls option as a dependency of `transport_encryption` so that we do not alter anything with the existing feature. With custom tls option as an dependency, I just needed to add code for custom tls, no modification of the default feature code.